wikitexthtml/render/preprocess.py

```diff
@@ -73,6 +85,44 @@ def hr_line(instance: WikiTextHtml, body: str) -> str:
     return regex.sub(r"\n----+\n", r"\n\n<hr />\n", body)
 
 
+def code_block(instance: WikiTextHtml, body: str) -> str:
+    # Use Pygments to generate highlighted code blocks
+
+    # The SYNTAXHIGHLIGHT_PATTERN Generates 6 groups:
+    #   The first group (match[0]) is the entire match
+    #   The third group (match[2]) is the language
+    #   The fourth group (match[3]) is a boolean to specify line numbering
+    #   The fifth group (match[4]) is the line number to start at
+    #   The sixth group (match[5]) is the code to highlight
+    # You can also reference the groups by name, i.e. match.group("lang")
```

Comment:
Feels a bit like you are explaining regex :P Anyway, as you never actually use the indexed method of accessing `match`, maybe just name them by name in the lines above, and leave out this line? That way I can match the code below to this comment block. Now I don't know if "lang" for example is actually [2] or not :)

Reply:
Good catch. I actually _did_ access by index originally, but added capture groups once I had about 4 different ones.

And yes, I'm absolutely explaining regex. It helps _me_ understand it if or when I have to touch this code again, because the subtleties fall out of my brain after about a day.

Reply:
Yeah, so it wasn't just me that was like: is he explaining regex? :P It is fine ofc :)